kube-aws-updater

```diff
@@ -0,0 +1,184 @@
+#!/bin/bash
+
+# Shell style guide: https://google.github.io/styleguide/shell.xml
+
+# ## (-) master and worker steps
+# ## (*) worker specific steps
+#
+# - label nodes
+# - disable scheduling
+# - double ASG size
+# - wait for 2x group
+# - stop ASG actions
+#   * delete docker-reigistry pod
+#   * wait for docker-registry to be up
+# - drain/terminate labelled nodes (sequentially)
+# - resize ASG to original size
+# - re-enable ASG actions
+
+set -e
+
+retire_time=$(date +"%Y-%m-%dT%H-%M-%SZ")
+
+# flags
+kube_context=''
+aws_profile=''
+role=''
+registry_flag=''
+
+usage() {
+  cat <<EOF
+Usage: $0 -c <kube_context> -p <aws_profile> -r [role] [-e]
+
+  -e    Registry flag - set if cluster contains docker registry and
+        needs to be moved prior to rest of the pods"
+EOF
+}
+
+while getopts 'c:p:r:eh' flag; do
+  case "${flag}" in
+    c) kube_context="${OPTARG}" ;;
+    p) aws_profile="${OPTARG}" ;;
+    r) role="${OPTARG}" ;;
+    e) registry_flag='true' ;;
+    h) usage && exit 0 ;;
+    *) echo "Unexpected option ${flag}" && usage && exit 1 ;;
+  esac
+done
+
+### Deps
+deps="
+  jq
+  kubectl
+  aws
+"
+
+missing_deps=
+for d in ${deps}; do
+  if ! command -v ${d} &> /dev/null; then
+    missing_deps+="${d} "
+  fi
+done
+
+if [[ -n "${missing_deps}" ]]; then
+  echo "Missing dependencies: ${missing_deps}"
+  exit 1
+fi
+
+###
+
+### Validation
+if [[ -z "${kube_context}" ]]; then
+  usage
+  exit 1
+fi
+
+aws_opts=
+if [[ -n "${aws_profile}" ]]; then
+  if aws configure list --profile ${aws_profile} &> /dev/null; then
+    aws_opts="--profile=${aws_profile}"
+    aws ${aws_opts} iam get-user >/dev/null || exit 1
```

Comment:
You prob want `&> /dev/null` here.

Reply:
What does that do?

Reply:
> &>/dev/null
This is just an abbreviation for >/dev/null 2>&1. It redirects file descriptor 2 (STDERR) and descriptor 1 (STDOUT) to /dev/null.

Reply:
Resolved